vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+This can occur because the sampler is having difficulty exploring the posterior parameter space given the model and your data.
+You can a few simple steps to assist with this:
+
+
+1. Increase the [`adapt_delta`](.adaptDelta) argument to 15 which will allow longer trajectories to used for steps when sampling. This is especially important if the `per_at_max_treedepth` value is high in the returned CmdStan diagnostics.
+2. Use more [informative priors](.settingPriors). 
+3. Try to simplify the model first to understand potential degeneracies.
+4. Reduce the amount of data you are trying to fit. This might include reducing the duration of time overwhich you are trying to fit. Once you can confirm that the model will fit your data, it may just be a question of computational power when fitting the larger data. 
+
+
+### Divergent transitions 
+
+CmdStan may alert you that you have ["divergent transitions"](https://mc-stan.org/docs/reference-manual/divergent-transitions.html) detected during your model fit.
+
+```r
+#>1: There were 32 divergent transitions after warmup.
+```
+
+A high number of divergent transitions indicates that the model outputs are not to be trusted.
+The first step in addressing this is to increase the [`adapt_delta`](#adaptDelta) argument to a numeric near one (e.g., 0.99 or 0.999).
+This will allow for a smaller step size and may result in more realiable inferences.
+Importantly, if the number of divergent transitions remains high, your model may be misspecified given the data that are available.
+At this point it may be worthwhile to investigate if the model paramterisation is correct for the data you have available.
+
+
+### My `r_hat`s are high and by `ess`s are low
+
+High values of `r_hat`, the Gelman-Rubin statistic used to measure converge, indicates a lack of convergence.
+Values greater than 1.05 are considered unreliable.
+Similarly, lower values for the [effective sample size](https://mc-stan.org/docs/reference-manual/effective-sample-size.html) is especially problematic.
+The effective sample size provides insight into how many independent samples you have drawn from the posterior, accounting for the autocorrelation within the Markov chains.
+Ideally you want these values to be near your total number of samples passed in the `iter_sample` argument.
+If `r_hat` is large and `ess_bulk`/ `ess_tail` are low, consider:
+
+1. Increasing your [adapt_delta](#adaptDelta) argument to a numeric value near 1 (e.g., 0.99). This will reduce the step size and could help with sampling.
+2. Use more [informative priors](.settingPriors). 
+
+
+### The posterior estimates are very wide
+
+Wide posterior values indicate that there is a high degree of uncertainity given the data and model.
+If not other warnings are shown (e.g., divergent transitions, higher ESS), then this is simply the inference given the available data.
+Posteriors could be reduced through the use of stronger, more informative [priors](#settingPriors).
```

Comment:
Dropping another plug for prior predictive sim here!

Reply:
@zsusswein

Reply:
Something along the lines of:

```suggestion
If no other warnings are shown (e.g., divergent transitions, low ESS), then this result is simply the inference given the available data and prior likelihood. 

However, Bayesian model building is often an iterative process [@gelmanBayesianWorkflow2020] and, in practice, it's not uncommon to realize that that the model's prior likelihood does accurately reflect your prior beliefs. 
Priors that were thought to be non-informative can turn out to [strongly influence the posterior](http://www.stat.columbia.edu/~gelman/research/published/entropy-19-00555-v2.pdf), often in unexpected and undesirable ways.
If you suspect that this might be the case, it can make sense to revisit your priors, iteratively inspecting the joint prior model through prior predictive simulation and tweaking marginal distributions over individual parameters.
The (Stan Prior Choice wiki)[https://github.com/stan-dev/stan/wiki/Prior-Choice-Recommendations] is often a useful guide to this process.
```